client/components/videos-ui/index.jsx

```diff
@@ -3,42 +3,37 @@ import { Button, Gridicon } from '@automattic/components';
 import classNames from 'classnames';
 import { useTranslate } from 'i18n-calypso';
 import moment from 'moment';
-import { useEffect, useState, useMemo } from 'react';
+import { useEffect, useState } from 'react';
 import Notice from 'calypso/components/notice';
-import useCourseQuery from 'calypso/data/courses/use-course-query';
-import useUpdateUserCourseProgressionMutation from 'calypso/data/courses/use-update-user-course-progression-mutation';
+import {
+	COURSE_SLUGS,
+	useCourseData,
+	useUpdateUserCourseProgressionMutation,
+} from 'calypso/data/courses';
 import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
 import VideoPlayer from './video-player';
 import './style.scss';
 
-const VideosUi = ( { HeaderBar, FooterBar, areVideosTranslated = true } ) => {
+const VideosUi = ( {
+	courseSlug = COURSE_SLUGS.BLOGGING_QUICK_START,
+	HeaderBar,
+	FooterBar,
+	areVideosTranslated = true,
+} ) => {
 	const translate = useTranslate();
 	const isEnglish = config( 'english_locales' ).includes( translate.localeSlug );
-
-	const courseSlug = 'blogging-quick-start';
-	const { data: course } = useCourseQuery( courseSlug, { retry: false } );
+	const { course, videoSlugs, completedVideoSlugs, isCourseComplete } = useCourseData( courseSlug );
 	const { updateUserCourseProgression } = useUpdateUserCourseProgressionMutation();
 
-	const initialUserCourseProgression = useMemo( () => course?.completions ?? [], [ course ] );
-
-	const [ userCourseProgression, setUserCourseProgression ] = useState( [] );
```

Comment:
We don't need initialUserCourseProgression and userCourseProgression anymore as the course would be updated whenever the video is marked as completed.